### PR TITLE
Remove QSIPrep-specific parameters and Config elements

### LINKED ISSUE #16: [BUG] output-resolution shouldn't be mandatory if --recon-input is used

### PR DESCRIPTION
**Pull request by tsalo**
Closes #16.

## Changes proposed in this pull request

- Remove the following CLI parameters and their associated config attributes:
    - `--skip-bids-validation`
    - `--anat-only` / `workflow.anat_only`
    - `--ignore` / `workflow.ignore`
    - `--skip-anat-based-spatial-normalization` / `execution.skip_anat_based_spatial_normalization`
    - `--anat-modality` / `workflow.anat_modality`
    - `--dwi-denoise-window` / `workflow.dwi_denoise_window`
    - `--denoise-method` / `workflow.denoise_method`
    - `--unringing-method` / `workflow.unringing_method`
    - `--dwi-no-biascorr` / `workflow.dwi_no_biascorr`
    - `--b1-biascorrect-stage` / `workflow.b1_biascorrect_stage`
    - `--no-b0-harmonization` / `workflow.no_b0_harmonization`
    - `--denoise-after-combining` / `workflow.denoise_after_combining`
    - `--separate-all-dwis` / `workflow.separate_all_dwis`
    - `--distortion-group-merge` / `workflow.distortion_group_merge`
    - `--anatomical-template` / `workflow.anatomical_template`
    - `--b0-to-t1w-transform` / `workflow.b0_to_t1w_transform`
    - `--intramodal-template-iters` / `workflow.intramodal_template_iters`
    - `--intramodal-template-transform` / `workflow.intramodal_template_transform`
    - `--b0-motion-corr-to` / `workflow.b0_motion_corr_to`
    - `--hmc-transform` / `workflow.hmc_transform`
    - `--hmc-model` / `workflow.hmc_model`
    - `--eddy-config` / `workflow.eddy_config`
    - `--shoreline-iters` / `workflow.shoreline_iters`
    - `--pepolar-method` / `workflow.pepolar_method`
    - `--fmap-bspline` / `workflow.fmap_bspline`
    - `--fmap-no-demean` / `workflow.fmap_demean`
    - `--use-syn-sdc` / `workflow.use_syn_sdc`
    - `--force-syn` / `workflow.force_syn`
    - `--recon-only` / `execution.recon_only`
    - `--recon-input` / `execution.recon_input`
- Remove the following config attributes:
    - `execution.output_layout`
    - `execution.qsirecon_dir`
    - `execution.run_preproc_and_recon`
    - `execution.task_id`
    - `workflow.cifti_output`
    - `workflow.do_reconall`
    - `workflow.dwi_only`
- Initially remove but then add back in:
    - `--b0-threshold` / `workflow.b0_threshold`
    - `--skip-odf-reports` / `execution.skip_odf_reports`
    - `--output-resolution` / `workflow.output_resolution`

### CODE REVIEW

File: qsirecon/config.py [hunk 1]
```diff
@@ -412,43 +412,20 @@ class execution(_Config):
...
-    output_layout = None
-    """Layout of derivatives within output_dir."""
-    # output_spaces = None
-    # """List of (non)standard spaces designated (with the ``--output-spaces`` flag of
-    # the command line) as spatial references for outputs."""
     reports_only = False
     """Only build the reports, based on the reportlets found in a cached working directory."""
     run_uuid = f"{strftime('%Y%m%d-%H%M%S')}_{uuid4()}"
     """Unique identifier of this particular run."""
-    skip_odf_reports = False
```
Comment (mattcieslak) [inline qsirecon/config.py:430]: We'll need to keep this one. This specifically skips creating reports that often cause crashes on hpc nodes without displays

File: qsirecon/config.py [hunk 2]
```diff
@@ -567,83 +541,14 @@ def _process_value(value):
...
-    anat_modality = None
-    """Modality to use as the anatomical reference. Images of this
-    contrast will be skull stripped and segmented for use in the
-    visual reports and reconstruction. If --infant, T2w is forced."""
-    anat_only = False
-    """Execute the anatomical postprocessing only."""
-    anatomical_template = None
-    """Keeps the :py:class:`~niworkflows.utils.spaces.SpatialReferences`
-    instance keeping standard and nonstandard spaces."""
-    b0_threshold = None
```
Comment (mattcieslak) [inline qsirecon/config.py:579]: we still need b0_threshold for some of the recon workflows

File: qsirecon/config.py [hunk 3]
```diff
@@ -567,83 +541,14 @@ def _process_value(value):
...
-    """Any value in the .bval file less than this will be considered a b=0 image."""
-    b0_motion_corr_to = None
-    """Perform SHORELine's initial b=0-based registration to first volume?
-    Or make a template? Either 'iterative' or 'first'"""
-    b0_to_t1w_transform = None
-    """Transformation model for intramodal registration."""
-    b1_biascorrect_stage = None
-    """The stage of processing at which to apply B1 bias correction. Either "final" (after
-    resampling), "none" (skipped entirely) or "legacy" (before concatenation)."""
-    cifti_output = None
```
Comment (mattcieslak) [inline qsirecon/config.py:589]: I could see a ScalarMapper writing to cifti sometime in the future

File: qsirecon/config.py [hunk 4]
```diff
@@ -567,83 +541,14 @@ def _process_value(value):
...
     """Configure pipelines specifically for infant brains"""
-    intramodal_template_iters = None
-    """Number of iterations for intramodal template construction."""
-    intramodal_template_transform = None
-    """Transformation used for building the intramodal template."""
     longitudinal = False
     """Run FreeSurfer ``recon-all`` with the ``-logitudinal`` flag."""
-    no_b0_harmonization = False
-    """Skip re-scaling dwi scans to have matching b=0 intensities."""
-    output_resolution = None
```
Comment (mattcieslak) [inline qsirecon/config.py:630]: I think ScalarMapper still needs this

File: qsirecon/cli/parser.py [hunk 5]
```diff
@@ -268,7 +219,6 @@ def _bids_filter(value, parser):
     )
 
     g_subset = parser.add_argument_group("Options for performing only a subset of the workflow")
-    g_subset.add_argument("--anat-only", action="store_true", help="Run anatomical workflows only")
     g_subset.add_argument(
```
Comment (mattcieslak) [inline qsirecon/cli/parser.py:222]: Can go too

File: qsirecon/cli/parser.py [hunk 6]
```diff
@@ -612,13 +296,6 @@ def _bids_filter(value, parser):
         type=Path,
         help="Directory containing freesurfer outputs to be integrated into recon.",
     )
-    g_recon.add_argument(
-        "--skip-odf-reports",
-        "--skip_odf_reports",
-        action="store_true",
-        default=False,
-        help="run only reconstruction, assumes preprocessing has already completed.",
-    )
```
Comment (mattcieslak) [inline qsirecon/cli/parser.py:621]: The help for this is incorrect - it should say that this skips 3d-rendered images, which often cause crashes when run on machines without monitor

Comment (mattcieslak) [reply qsirecon/cli/parser.py:621]: And it should be kept. Lots of HPCs struggle to make the odf reports

File: qsirecon/cli/parser.py [hunk 7]
```diff
@@ -315,157 +255,6 @@ def _bids_filter(value, parser):
...
-    g_conf.add_argument(
-        "--b0-threshold",
-        "--b0_threshold",
-        action="store",
-        type=int,
-        default=100,
-        help="any value in the .bval file less than this will be considered "
-        "a b=0 image. Current default threshold = 100; this threshold can be "
-        "lowered or increased. Note, setting this too high can result in inaccurate results.",
-    )
```
Comment (mattcieslak) [inline qsirecon/cli/parser.py:343]: We'll need to keep this argument

**Review by mattcieslak — CHANGES_REQUESTED** [review]
A couple unintuitive options have to be added back but otherwise looks perfect

File: qsirecon/config.py (hunk 3 above)
Comment (tsalo) [reply qsirecon/config.py:589]: I'm hoping that, by that time, we'll have CIFTIs defined in `--output-spaces` instead.

File: qsirecon/config.py (hunk 4 above)
Comment (tsalo) [reply qsirecon/config.py:630]: It doesn't seem to be used in #11

Comment (tsalo) [reply qsirecon/config.py:630]: I see now that it's because I removed `init_output_grid_wf`, but I shouldn't have. I've added it back in.

**Review by mattcieslak — APPROVED** [review]

**Pull request merged**